packages/turbo-repository/rust/src/lib.rs

```diff
@@ -35,18 +42,47 @@ pub struct Workspace {
     /// The package manager used by the workspace.
     #[napi(readonly)]
     pub package_manager: PackageManager,
+    /// The package graph for the workspace.
+    graph: PackageGraph,
 }
 
+#[napi]
 impl Package {
-    fn new(workspace_path: &AbsoluteSystemPath, package_path: &AbsoluteSystemPath) -> Self {
+    fn new(
+        name: String,
+        workspace_path: &AbsoluteSystemPath,
+        package_path: &AbsoluteSystemPath,
+    ) -> Self {
         let relative_path = workspace_path
             .anchor(package_path)
             .expect("Package path is within the workspace");
         Self {
+            name,
             absolute_path: package_path.to_string(),
             relative_path: relative_path.to_string(),
         }
     }
+
+    fn dependents(
+        &self,
+        graph: &PackageGraph,
+        workspace_path: &AbsoluteSystemPath,
+    ) -> Vec<Package> {
+        let node = WorkspaceNode::Workspace(WorkspaceName::Other(self.name.clone()));
+        let ancestors = graph.immediate_ancestors(&node).unwrap();
```

Comment:
We should do our best to avoid panic even if it's a user error. If a user passes in a `graph` that doesn't contain this `Package` this will result in a panic.

Reply:
This is a private function though, and the only user is coming from an internal implementation of find_packages() which should get a list of valid packages. I can be more defensive here, but does that ease the concern?

Reply:
updated to return empty Vec in this case